modules/caddyhttp/templates/tplcontext.go

```diff
@@ -183,20 +183,28 @@ func (c TemplateContext) funcHTTPInclude(uri string) (string, error) {
 	buf.Reset()
 	defer bufPool.Put(buf)
 
+	// Create a new virtual request
 	virtReq, err := http.NewRequest("GET", uri, nil)
 	if err != nil {
 		return "", err
 	}
+
+	// Set the Host header to the original request's Host
 	virtReq.Host = c.Req.Host
+
+	virtReq.RemoteAddr = "127.0.0.1"
+
+	// Clone the original request's headers and add X-Forwarded-For
 	virtReq.Header = c.Req.Header.Clone()
-	virtReq.Header.Set("Accept-Encoding", "identity") // https://github.com/caddyserver/caddy/issues/4352
+	virtReq.Header.Set("Accept-Encoding", "identity")
```

Comment:
Keep the comment as-is, the link to the issue is important to explain the context for this line being in place
```suggestion
	virtReq.Header.Set("Accept-Encoding", "identity") // https://github.com/caddyserver/caddy/issues/4352
```